sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+
+        TODO: Add examples
+
+        """
+        return Predicate('antihermitian')
+
+    @memoize_property(predicate_storage)
+    def real(self):
+        r"""
+        Real number predicate.
+
+        ``Q.real(x)`` is true iff ``x`` is a real number, i.e., it is in the
+        interval :math:`(-\infty, \infty)`.  Note that, in particular the infinities
+        are not real. Use ``Q.extended_real`` if you want to consider those as well.
+
+        A few important facts about reals:
+
+        - Every real number is positive, negative, or zero.  Furthermore, because
+          these sets are pairwise disjoint, each real number is exactly one of
+          those three.
+
+        - Every real number is also complex.
+
+        - Every real number is either rational or irrational.
+
+        - Every real number is either algebraic or transcendental.
+
+        - The facts ``Q.negative``, ``Q.zero``, ``Q.positive``, ``Q.nonnegative``,
+          ``Q.nonpositive``, ``Q.nonzero``, ``Q.integer``, ``Q.rational``, and
+          ``Q.irrational`` all imply ``Q.real``, as do all facts that imply those
+          facts.
+
+        - The facts ``Q.algebraic``, and ``Q.transcendental`` do not imply
+          ``Q.real``; they imply ``Q.complex``. An algebraic or transcendental
+          number may or may not be real.
+
+        - The "non" facts (i.e., ``Q.nonnegative``, ``Q.nonzero``, and
+          ``Q.nonpositive``) are not equivalent to not the fact, but rather, not
+          the fact *and* ``Q.real``.  For example, ``Q.nonnegative`` means
+          ``~Q.negative & Q.real``. So for example, ``I`` is not nonnegative,
+          nonzero, or nonpositive.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols
+        >>> x = symbols('x')
+        >>> ask(Q.real(x), Q.positive(x))
+        True
+        >>> ask(Q.real(0))
+        True
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Real_number
+
+        """
+        return Predicate('real')
+
+    @memoize_property(predicate_storage)
+    def extended_real(self):
+        r"""
+        Extended real predicate.
+
+        ``Q.extended_real(x)`` is true iff ``x`` is a real number including
+        :math:`{-\infty, \infty}`.
+
+        See documentation of ``Q.real`` for more information about related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, oo, I
+        >>> ask(Q.extended_real(1))
+        True
+        >>> ask(Q.extended_real(I))
+        False
+        >>> ask(Q.extended_real(oo))
+        True
+
+        """
+        return Predicate('extended_real')
+
+    @memoize_property(predicate_storage)
+    def imaginary(self):
+        """
+        Imaginary number predicate.
+
+        ``Q.imaginary(x)`` is true iff ``x`` can be written as a real
+        number multiplied by the imaginary unit ``I``. Please note that ``0``
+        is not considered to be an imaginary number.
```

Comment:
I hope we change this soon. These two sentences directly contradict one another.